src/main/java/net/sf/jabref/logic/util/io/SimpleFileListEntry.java

```diff
@@ -0,0 +1,94 @@
+/*  Copyright (C) 2015 JabRef contributors.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.logic.util.io;
+
+import java.util.List;
+
+/**
+ * This class represents a simplified version of the file link for a Bibtex entry.
+ */
+public class SimpleFileListEntry {
```

Comment:
I do not like the current implementation of FileListEntry, SimpleFileListEntry and SimpleFileList. 

The thing is that the current implementation suffers as it comes directly from dependency hell. ;)

What I would expect to be there as classes:
- ExternalFileType defines the type of the ExternalFile (currently named FileListEntry). A class (could be bibenty, but I am not sure if it should not be another class) should be able to decode a List<ExternalFile> from its file attribute. 
- ExternalFileType should be independent from the GUI, and then the ExternalFile would be GUI independent as well - then both ExternalFile and ExternalFileType can be put in the model. At the moment ExternalFileType contains a label and an icon field. Hence, this should be split up into a GUIDependentExternalFileType which contains both the JLabel and the Icon plus a reference to the ExternalFileType which contains all the information which is serialized. Then, the SimpleFileListEntry is not necessary anymore. The issue here is that the current mapping of an ExternalFileType to an Icon is currently hard-coded, so we need a large switch statement in the convertion code of an ExternalFileType to a GUIDependentExternalFileType.